apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -305,9 +305,6 @@ const aichatSlice = createSlice({
     setViewMode: (state, action: PayloadAction<ViewMode>) => {
       state.viewMode = action.payload;
     },
-    setHasPublished: (state, action: PayloadAction<boolean>) => {
-      state.hasPublished = action.payload;
-    },
     setStartingAiCustomizations: (
       state,
       action: PayloadAction<{
```

Comment:
Just trying to remember - are we setting `levelAichatSettings` as optional because we've refactored `aichat` levels so some of the older levels may not have this defined?

Reply:
Once `aichat` .level settings are stable, we'll probably need to do an audit of the .level files. Seems like we'd want `levelAichatSettings` to always be present. Thanks!

Reply:
`levelAichatSettings` are optional only because `levelProperties` can technically be undefined in lab2 - that said though I agree with making them required in this function at least, and moving the check up into AichatView, since for Aichat levels, they should always be defined.